design/design-doc-packages-v2.md

```diff
@@ -0,0 +1,1067 @@
+  // it is not known. Either Package or CustomResourceDefinition can be
+  // specified.
+  CustomResourceDefinition string `json:"crd,omitempty"`
+}
+
+// PackageControllerOptions allow for changes in the Package extraction and
+// deployment controllers. These can affect how images are fetched and how
+// Package derived resources are created.
+type PackageControllerOptions struct {
+  // ImagePullSecrets are named secrets in the same workspace that can be used
+  // to fetch Packages from private repositories and to run controllers from
+  // private repositories
+  ImagePullSecrets []corev1.LocalObjectReference `json:"imagePullSecrets,omitempty"`
+
+  // ImagePullPolicy defines the pull policy for all images used during
+  // Package extraction and when running the Package controller.
+  // https://kubernetes.io/docs/concepts/configuration/overview/#container-images
+  ImagePullPolicy corev1.PullPolicy `json:"imagePullPolicy,omitempty"`
+
+  // ServiceAccount options allow for changes to the ServiceAccount the
+  // Package Manager creates for the Package's controller
+  ServiceAccount *ServiceAccountOptions `json:"serviceAccount,omitempty"`
+}
+
+// ServiceAccountOptions augment the ServiceAccount created by the Package
+// controller
+type ServiceAccountOptions struct {
+  Annotations map[string]string `json:"annotations,omitempty"`
+}
+
+// PackageInstallStatus represents the observed state of a PackageInstall.
+type PackageInstallStatus struct {
+  runtimev1alpha1.ConditionedStatus `json:"conditionedStatus,omitempty"`
+
+  InstallJob    *corev1.ObjectReference `json:"installJob,omitempty"`
+  PackageRecord *corev1.ObjectReference `json:"packageRecord,omitempty"`
+}
+```
+
+The following changes will be made to the `ClusterPackageInstall` schema.
+
+Fields to be added:
+
+- `spec.revisionActivationPolicy`: specifies how a package should update to a
+  new version. Options are `Automatic` or `Manual`. If set to `Manual`,
+  `PackageRevision`s will be created, but must be manually activated. Defaults
+  to `Automatic`.
+- `spec.revisionHistoryLimit`: number of `Inactive` `PackageRevision`s to keep.
+  Once a `PackageRevision` is outside of this range it and all of the objects
+  that only it owns will be cleaned up. Defaults to 1. If set to 0,
+  `PackageRevision`s will not be garbage collected.
+- `status.currentRevision`: a string that is the full representation of the
+  currently `Active` package image.
+
+Fields to be removed:
+
+- `spec.source`: the purpose of this field was to override the source of the
+  packaged controller image. Packages are meant to specify the exact controller
+  they are bringing, this should not be customizable.
+- `status.installJob`: a reference to the install `Job` for the
+  `ClusterPackageInstall`.
+- `status.packageRecord`: a reference to the `Package` that represents the
+  installation unit for the `ClusterPackageInstall`. The `PackageRevision` is
+  now discoverable by name at the cluster scope.
```

Comment:
@negz in the section above it mentions the `status.currentRevision` field will be added. It is a string instead of an object reference and will match the hash of the image, which is also the name of the `PackageRevision`. This will be used to get the `PackageRevision` at the cluster scope by name.